lib/scss_lint/linter/groupable_selector.rb

```diff
@@ -0,0 +1,64 @@
+# add missing comparison for literals
+module Sass::Script::Tree
+  class Literal
+
+    def == other
+      self.class == other.class and self.value == other.value
+    end
+  end
+end
+
+module SCSSLint
+  class Linter::GroupableSelector < Linter
+    include LinterRegistry
+
+    def check_node(node)
+      visits = Set.new
+      node.children.each do |child_node|
+        groupable_node = find_groupable_node child_node, node.children
+        next unless groupable_node
+        next if visits.include? [child_node, groupable_node] or
+                visits.include? [groupable_node, child_node]
+        visits << [child_node, groupable_node]
+
+        add_lint child_node.line, "Group identical rule `#{node_rule child_node}` on line #{child_node.line} " \
+        "with rule `#{node_rule groupable_node}` on line #{groupable_node.line}" do
+          node.children.delete groupable_node
+          child_node.parsed_rules = nil
```

Comment:
Does this do what you think it does? There are multiple related fields defined on `RuleNode`: `rule`, `parsed_rules`, `resolved_rules`, etc. It makes me uncomfortable that we're relying on the behavior of this specific rule and hoping it will continue to work the same way in future.